src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/NcksSubsetParameters.java

```diff
@@ -11,9 +16,13 @@ public void addTimeSubset(String netcdfVariableName, Subset timeSubset) {
         // ncks works in microseconds and time values passed to gogoduck by the portal
         // are in millisecond precision thanks to java/javascript date usage in harvesting/thredds/
         // portal code.
-        // To ensure the truncated millisecond values passed select the untruncated microsecond values
-        // they were derived from add the maximum possible truncation error to the end time value
-        Subset ncksTimeSubset = new Subset(timeSubset.start, timeSubset.end.replaceAll("(\\.\\d\\d\\d)Z", "$1999Z"));
+        // To ensure the rounded millisecond values passed select the unrounded microsecond values
+        // they were derived from add the possible millisecond rounding or truncation error to the from and to times.
+        CalendarDate start = CalendarDateFormatter.isoStringToCalendarDate(Calendar.gregorian, timeSubset.start);
```

Comment:
Are the dates held in floating point somewhere, which creates rounding errors in truncation?